v2/internal/resolver/resource_hierarchy.go

```diff
@@ -122,23 +130,76 @@ func (h ResourceHierarchy) FullyQualifiedARMID(subscriptionID string) (string, e
 		// Join them together
 		interleaved := genruntime.InterleaveStrSlice(resourceTypes, remainingNames)
 		return genericarmclient.MakeResourceGroupScopeARMID(subscriptionID, rgName, provider, interleaved...)
+	case ResourceHierarchyRootTenant:
+		// The only resource we actually care about for figuring out resource types is the
+		// most derived resource
+		res := h[len(h)-1]
```

Comment:
could this run into indexOutOfBounds if `len==0`?

Reply:
ok, makes sense .. it panics in `rootKind` method if `len==0`

Reply:
It could, yes. There are other places where we explicitly panic on `len==0`, but basically a `ResourceHierarchy` is expected to be at least length = 1. Moreover, these are created by functions we control and they always have at least the resource being deployed in them, so there's no way in practice to end up with 1 that's length = 0.

Good question though.